src/components/Board.vue

```diff
@@ -0,0 +1,472 @@
+  // data
+  width = 10;
+
+  fieldSizeMin = 5;
+
+  fieldSizeMax = 20;
+
+  bombsMin = 5;
+
+  bombsMax = 40;
+
+  isGameOver = false;
+
+  timerEnabled = true;
+
+  squares: Array<string> = [];
+
+  cells: Array<any> = []; // HTMLDivElements array
+
+  gameState: string = States.Start;
+
+  topEdge: number = this.width - 1;
+
+  topLeftCorner: number = this.width;
+
+  lastCell: number = this.width * this.width - 1;
+
+  downEdge: number = this.width * this.width - this.width;
+
+  downRightCorner: number = this.width * this.width - this.width - 1;
+
+  classesToDelete: Array<string> = [
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "bomb",
+    "flag",
+    "checked"
+  ];
+
+  // store
+  public updateBombs(e): void {
+    this.$store.commit("updateBombs", parseInt(e.target.value));
+  }
+
+  get flags(): number {
+    const { flags } = this.$store.state;
+    return flags;
+  }
+
+  updateFlags(): void {
+    this.$store.dispatch("updateFlags", this.flags);
+  }
+
+  // watchers
+  @Watch("width")
+  onWidthChange(width: number) {
+    this.topEdge = width - 1;
+    this.topLeftCorner = width;
+    this.lastCell = width * width - 1;
+    this.downEdge = width * width - width;
+    this.downRightCorner = width * width - width - 1;
+  }
+
+  // methods
+  private async prepareNewGame() {
+    const grid = document.querySelector<HTMLElement>(".grid");
+    if (this.$store.state.bombsAmount < this.width * this.width) {
+      this.$store.dispatch("updateBombs", this.$store.state.bombsAmount);
+    } else {
+      alert(`You can't set more bombs than fields.`);
+      this.$store.dispatch("updateBombs", 10);
+    }
+    if (grid != null) {
+      grid.style.width = `${this.width * 40}px`;
+      await this.clearHelper();
+      await this.createBoard();
+      await this.fillCells();
+      this.stopTimer();
+      this.resetTimer();
+      this.timerEnabled = true;
+    }
+  }
+
+  private createBoard(): void {
+    const bombsArray = Array(this.$store.state.bombsAmount).fill("bomb");
+    const emptyArray = Array(
+      this.width * this.width - this.$store.state.bombsAmount
+    ).fill("empty");
+    const gameArray = emptyArray.concat(bombsArray);
+    const shuffledArray = gameArray.sort(() => Math.random() - 0.5);
+    this.squares = [...shuffledArray];
+  }
+
+  private fillCells(): void {
+    const elements = document.getElementsByClassName("square");
+    for (let i = 0; i < elements.length; i++) {
+      this.cells.push(elements[i]);
+    }
+    this.addNumbers();
+  }
+
+  public clicked(square) {
+    if (this.timerEnabled === true) {
+      this.resetTimer();
+      this.startTimer();
+      this.timerEnabled = false;
+    }
+    if (square === undefined) {
+      square = window.event!.target;
+    }
+    const currentId = square.id;
+    if (this.isGameOver) return;
+    if (
+      square.classList.contains("checked") ||
+      square.classList.contains("flag")
```

Comment:
nie umieszczaj stanu gry w html'u. Html jest odzwierciedleniem stanu gry. Stwórz dwuwymiarową tablicę odzwierciedlającą stan gry. Zastanów się jakie wartości ta tablica powinna przechowywać. Ten stan gry musisz potem zapisać w db albo w storze a w jaki sposób to zrobić z htmlem?